jetty-ee10/jetty-ee10-demos/jetty-ee10-demo-spec/jetty-ee10-demo-spec-webapp/src/test/java/org/eclipse/jetty/ee10/demos/SpecWebAppTest.java

```diff
@@ -0,0 +1,155 @@
+    @BeforeEach
+    public void setup(WorkDir workDir) throws Exception
+    {
+        server = new Server();
+
+        ServerConnector connector = new ServerConnector(server);
+        connector.setPort(0);
+        server.addConnector(connector);
+
+        Path webappDir = prepareWebAppDir(workDir);
+
+        WebAppContext webapp = new WebAppContext();
+        ResourceFactory resourceFactory = ResourceFactory.of(webapp);
+        webapp.setContextPath("/");
+        webapp.setWarResource(resourceFactory.newResource(webappDir));
+        webapp.setAttribute(
+            "org.eclipse.jetty.server.webapp.ContainerIncludeJarPattern",
+            ".*/jakarta.servlet-api-[^/]*\\.jar$|.*/[^/]*taglibs.*\\.jar$");
+
+        HashLoginService hashLoginService = new HashLoginService();
+        hashLoginService.setName("Test Realm");
+        Path realmFile = MavenPaths.findTestResourceFile("ee10-demo-realm.properties");
+        Resource realmResource = ResourceFactory.of(server).newResource(realmFile);
+        hashLoginService.setConfig(realmResource);
+        SecurityHandler securityHandler = webapp.getSecurityHandler();
+        securityHandler.setLoginService(hashLoginService);
+
+        new org.eclipse.jetty.ee10.plus.jndi.Resource(webapp, "jdbc/mydatasource", new MockDataSource());
+        new org.eclipse.jetty.ee10.plus.jndi.Transaction("ee10", new MockUserTransaction());
+
+        server.setHandler(webapp);
+        server.start();
+
+        client = new HttpClient();
+        client.start();
+    }
+
+    private Path prepareWebAppDir(WorkDir workDir) throws IOException
+    {
+        Path webappDir = workDir.getEmptyPathDir();
+        Path srcWebapp = MavenPaths.projectBase().resolve("src/main/webapp");
+        IO.copyDir(srcWebapp, webappDir);
+
+        Path webappClassesDir = webappDir.resolve("WEB-INF/classes");
+        FS.ensureDirExists(webappClassesDir);
+        Path classesDir = MavenPaths.projectBase().resolve("target/classes");
+        IO.copyDir(classesDir, webappClassesDir);
+
+        Path libDir = webappDir.resolve("WEB-INF/lib");
+        FS.ensureDirExists(libDir);
+        copyDependency("jetty-ee10-demo-container-initializer", libDir);
+        copyDependency("jetty-ee10-demo-web-fragment", libDir);
+
+        return webappDir;
+    }
+
+    private void copyDependency(String depName, Path libDir) throws IOException
+    {
+        Path depPath = MavenPaths.projectBase().resolve("../" + depName).normalize();
+        if (!Files.isDirectory(depPath))
+            fail("Dependency not found: " + depPath);
+        Path outputJar = libDir.resolve(depName + ".jar");
+        Map<String, String> env = new HashMap<>();
+        env.put("create", "true");
+
+        URI uri = URI.create("jar:" + outputJar.toUri().toASCIIString());
+        try (FileSystem fs = FileSystems.newFileSystem(uri, env))
+        {
+            Path root = fs.getPath("/");
+            IO.copyDir(depPath.resolve("target/classes"), root);
```

Comment:
uhm this is hackhish as it assume you are running full build

Reply:
fixed in a branch with this change https://github.com/eclipse/jetty.project/pull/9871/commits/b37aacff23a393b8afad604bbb4ba86df0dfe550

Reply:
Submit  PR with that change.